tests/e2e/traffic_splitting_fallback_test.go

```diff
@@ -0,0 +1,132 @@
+	const manifest = "testdata/traffic_splitting_fallback.yaml"
+	require.NoError(t, kubectlApplyManifest(t.Context(), manifest))
+
+	const egSelector = "gateway.envoyproxy.io/owning-gateway-name=traffic-splitting-fallback"
+
+	// Wait for the gateway to be ready.
+	requireWaitForGatewayPodReady(t, egSelector)
+
+	t.Run("traffic-distribution", func(t *testing.T) {
+		// Test that traffic splitting configuration is working by making multiple requests
+		// and verifying they are distributed between backends A and B.
+		const requestCount = 50
+		backendAResponses := 0
+		backendBResponses := 0
+
+		for i := 0; i < requestCount; i++ {
+			fwd := requireNewHTTPPortForwarder(t, egNamespace, egSelector, egDefaultServicePort)
+			defer fwd.kill()
+
+			req, err := http.NewRequest(http.MethodPost, fwd.address()+"/v1/chat/completions", strings.NewReader(
+				`{"messages":[{"role":"user","content":"Say this is a test"}],"model":"model-a"}`))
+			require.NoError(t, err)
+			req.Header.Set("x-ai-eg-model", "model-a")
+
+			// Use a generic response that will be overridden by the testupstream server.
+			req.Header.Set(testupstreamlib.ResponseBodyHeaderKey, base64.StdEncoding.EncodeToString([]byte(`{"choices":[{"message":{"content":"test"}}]}`)))
+
+			resp, err := http.DefaultClient.Do(req)
+			require.NoError(t, err)
+			defer resp.Body.Close()
+			require.Equal(t, 200, resp.StatusCode)
+
+			// Check response headers for backend identification.
+			backendID := resp.Header.Get("testupstream-id")
+			switch backendID {
+			case "backend-a":
+				backendAResponses++
+			case "backend-b":
+				backendBResponses++
+			default:
+				t.Logf("unexpected backend ID: %s", backendID)
+			}
+		}
+
+		// Validate that both backends received traffic (with reasonable tolerance for 50/50 split).
+		require.Positive(t, backendAResponses, "Backend A should receive some traffic")
+		require.Positive(t, backendBResponses, "Backend B should receive some traffic")
+		require.Equal(t, requestCount, backendAResponses+backendBResponses, "All requests should be handled")
+
+		// Check that distribution is roughly 50/50 (within 20% tolerance).
+		backendARatio := float64(backendAResponses) / float64(requestCount)
+		backendBRatio := float64(backendBResponses) / float64(requestCount)
+
+		// Validate that both backends receive traffic within 50/50 distribution (30%-70% tolerance).
+		require.Greater(t, backendARatio, 0.3, "Backend A should receive at least 30% of traffic")
+		require.Less(t, backendARatio, 0.7, "Backend A should receive at most 70% of traffic")
+		require.Greater(t, backendBRatio, 0.3, "Backend B should receive at least 30% of traffic")
+		require.Less(t, backendBRatio, 0.7, "Backend B should receive at most 70% of traffic")
+
+		t.Logf("Traffic distribution: Backend A=%d (%.1f%%), Backend B=%d (%.1f%%)",
+			backendAResponses, backendARatio*100, backendBResponses, backendBRatio*100)
+	})
+
+	t.Run("fallback-backend-c", func(t *testing.T) {
```

Comment:
this case we shouldn't get any response from backend-a or -b right? can we add more assertions to verify that?